google-cloud-examples/src/main/java/com/google/cloud/examples/bigtable/HelloWorld.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2018 Google LLC.  All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.examples.bigtable;
+
+import com.google.api.gax.rpc.ServerStream;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminClient;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminSettings;
+import com.google.cloud.bigtable.admin.v2.models.CreateTableRequest;
+import com.google.cloud.bigtable.admin.v2.models.Table;
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.models.InstanceName;
+import com.google.cloud.bigtable.data.v2.models.Query;
+import com.google.cloud.bigtable.data.v2.models.Row;
+import com.google.cloud.bigtable.data.v2.models.RowCell;
+import com.google.cloud.bigtable.data.v2.models.RowMutation;
+
+public class HelloWorld {
+
+  public static void main(String... args) throws Exception {
```

Comment:
I think it would be nicer to:
- add instance variables to HelloWorld for project id/instance id/etc
- make all methods non static
- remove per method parameters
- use main to simply instantiate HelloWord passing it the appropriate parameters and then call run

something like:
```java
public class HelloWorld {
  final String COLUMN_FAMILY_ID = "cf1";
  final String COLUMN_QUALIFIER = "greeting";
  final String TABLE_PREFIX = "table";
  //...

  private final String projectId;
  private final String instanceId;
  //...
  private final BigtableDataClient dataClient;
  private final BigtableTableClient tableAdminClient;

  public static void main(String[] args) {
    if (args.length != 3) 
     //..
    }
   String projectId = args[0];
   String instanceId = args[1];
   String tableId = args[2];
   HelloWorld helloWorld = new HelloWorld(projectId, instanceId, tableId);
   helloWorld.run();
  }

  public HelloWorld(String projectId, String instanceId, ...) {
    this.projectId = projectId;
    //..
   dataClient = BigtableDataClient.create(settings);
  }
  public void run() {
    createTable();
    writeToTable();
    //...
  }
  void createTable() {
    CreateTableRequest request = //...
    adminClient.createTable(request);
  }
  //...
}
```

Reply:
Should TableAdmin and InstanceAdmin samples be restructured like this as well?

Reply:
I think so, but can you do this for HelloWorld to start with and we can see how well it works out first?

Reply:
sure